src/workers/atlas_worker.js

```diff
@@ -8,20 +8,38 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
 */
 
 import PeriodicWorker from './periodic_worker';
+import ChallengeResolutionStrategy from './atlas_strategies/challenge_resolution_strategy';
 
 export default class AtlasWorker extends PeriodicWorker {
-  constructor(web3, dataModelEngine, challengesRepository, logger) {
-    super(5000, logger);
+  constructor(web3, dataModelEngine, challengesRepository, strategy, logger) {
+    super(strategy.challengePullingInterval, logger);
     this.web3 = web3;
     this.dataModelEngine = dataModelEngine;
+    this.strategy = strategy;
     this.challengesRepository = challengesRepository;
+    if (!(this.strategy instanceof ChallengeResolutionStrategy)) {
+      throw new Error('A valid strategy must be provided');
+    }
+  }
+
+  async resolve(bundle, challengeId) {
+    if (await this.strategy.shouldResolveChallenge(bundle)) {
+      await this.challengesRepository.resolveChallenge(challengeId);
+      await this.strategy.afterChallengeResolution(bundle);
+      this.logger.info({message: '🍾 Yahoo! The bundle is ours.', bundleId: bundle.bundleId});
+    } else {
+      this.logger.info({message: 'Challenge resolution cancelled', challengeId});
+    }
   }
 
   async tryToResolve({sheltererId, bundleId, challengeId}) {
-    this.logger.info({message: `Trying to fetch the bundle`, sheltererId, bundleId, challengeId});
-    await this.dataModelEngine.downloadBundle(bundleId, sheltererId);
-    await this.challengesRepository.resolveChallenge(challengeId);
-    this.logger.info({message: '🍾 Yahoo! The bundle is ours.', bundleId});
+    if (await this.strategy.shouldFetchBundle({sheltererId, bundleId, challengeId})) {
```

Comment:
You provide here parameters that are not declared in the interface of the ChallengeResolutionStrategy.

Reply:
Because if we declare arguments in abstract method the linter becomes angry